### ISSUE TITLE
Debug info lost when debugging complex code in computation expression

### ISSUE THREAD

**Issue by cartermp** [dotnet/fsharp#11262]
Repro:

1. Open VisualFSharp.sln
2. Open SignatureHelp.fs
3. Put a breakoint on line 123(`let startOfArgs = ...`)
4. Launch a hive of VS where you can debug
5. Enter the following source code:

```fsharp
type C() =
    static member M(x, y, z)  = ()

C.M(1)
```

6. Place cursor after the `1` on the last line
7. Press the comma character (`,`) so that Signature Help is triggered

Note that there are some locals, and critically you can observe the value of `applicableSpan`:

![image](https://user-images.githubusercontent.com/6309070/111529197-1a4e3100-871f-11eb-9202-f91d4e7f261c.png)

Now try to get the source code that matches this span by:

1. Opening the Immediate Window
2. Entering the following:

`sourceText.GetSubText(applicableSpan)`

You will observe that `sourceText` doesn't exist. Indeed, it wasn't even in the locals window.

![image](https://user-images.githubusercontent.com/6309070/111529367-4e295680-871f-11eb-93f8-d05e810b319a.png)

However, note that `sourceText` is a parameter to this method and it is clearly accessible.

We are losing debug info on every `do!` call in the method.

To test this, place a breakpoint on line 89 (`do! Option.guard (methods.Length > 0 ...`). When it is broken there, you'll see that `sourceText` is available:

![image](https://user-images.githubusercontent.com/6309070/111529603-96487900-871f-11eb-954d-2916ac5eb7d5.png)

**Comment by dsyme** [comment]
This is a general problem with loss of uncaptured variables when debugging F# closures or computation expressions.

In debug mode we should consider simply copying all values into the closure/continuation even if they are not captured.

**Comment by cartermp** [comment]
Yes, absolutely. This would make debugging FSharp.Editor.dll (and other CE-heavy things) so, so much better.
